.eslintrc.js

```diff
@@ -354,6 +363,7 @@ module.exports = {
         'react/require-default-props': 'off',
         'react/state-in-constructor': 'off',
         'react/static-property-placement': 'off',
+        'react/function-component-definition': 'off',
```

Comment:
Doesn't #34644 fix violations of this rule?

Reply:
These files use block scope to organize code, it conflicts with the `no-inner-declarations` rule. I'll disable that rule instead.

Reply:
ah, that's not possible because
```
Function declarations are not allowed inside blocks in strict mode when targeting 'ES3' or 'ES5'. Modules are automatically in strict mode.
```

Reply:
@michaldudak Ah, I'm seeing the confusion now, I'm disabling this rule only for `.spec.js` files here. The fixes in https://github.com/mui/material-ui/pull/34644 are in other files.